master/internal/task/idle_timeout.go

```diff
@@ -0,0 +1,57 @@
+package task
+
+import (
+	"time"
+
+	"github.com/determined-ai/determined/master/internal/proxy"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/actor/actors"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const tickInterval = 5 * time.Second
+
+// ProxyIdleTimeoutWatcherTick is the incoming message that should be handled.
+type ProxyIdleTimeoutWatcherTick struct{}
+
+// ProxyIdleTimeoutWatcher watches the proxy activity to handle a task actor idle timeout.
+type ProxyIdleTimeoutWatcher struct {
+	proxyRef *actor.Ref
+
+	TaskID      string
+	Description string
+	Timeout     model.Duration
+	KillMessage actor.Message
+}
+
+// PreStart should be called on task actor PreStart.
+func (p *ProxyIdleTimeoutWatcher) PreStart(ctx *actor.Context) {
+	p.proxyRef = ctx.Self().System().Get(actor.Addr("proxy"))
```

Comment:
can you put `actor.Addr("proxy")` into a pkg somewhere as a `var proxyAddr = ...` at the top. I think we should do this for all "well known" addresses (rm, trial logger, etc)

Reply:
or do it later/make a ticket for this pattern if you agree with it

Reply:
agreed. Will do that when we refactor interactive jobs.